packages/cli-platform-apple/src/commands/runCommand/getBuildPath.ts

```diff
@@ -42,63 +17,13 @@ export async function getBuildPath(
     throw new CLIError('Failed to get the app name.');
   }
 
-  return `${targetBuildDir}${
-    isCatalyst ? '-maccatalyst' : ''
-  }/${executableFolderPath}`;
-}
-
-async function getTargetPaths(
-  buildSettings: string,
-  scheme: string,
-  target: string | undefined,
-) {
-  const settings = JSON.parse(buildSettings);
-
-  const targets = settings.map(
-    ({target: settingsTarget}: any) => settingsTarget,
-  );
-
-  let selectedTarget = targets[0];
-
-  if (target) {
-    if (!targets.includes(target)) {
-      logger.info(
-        `Target ${chalk.bold(target)} not found for scheme ${chalk.bold(
-          scheme,
-        )}, automatically selected target ${chalk.bold(selectedTarget)}`,
-      );
-    } else {
-      selectedTarget = target;
-    }
-  }
-
-  // Find app in all building settings - look for WRAPPER_EXTENSION: 'app',
-
-  const targetIndex = targets.indexOf(selectedTarget);
-
-  const wrapperExtension =
-    settings[targetIndex].buildSettings.WRAPPER_EXTENSION;
-
-  if (wrapperExtension === 'app') {
-    return {
-      targetBuildDir: settings[targetIndex].buildSettings.TARGET_BUILD_DIR,
-      executableFolderPath:
-        settings[targetIndex].buildSettings.EXECUTABLE_FOLDER_PATH,
-    };
+  if (!fullProductName) {
+    throw new CLIError('Failed to get product name.');
```

Comment:
Let's leave this is for a later refactor, but these errors for missing builds ettings don't guide user what to do next and how to recover